### PR TITLE
feat: updated link chooser 

### PR DESCRIPTION
**Pull request by SamCreasey**
Updated link chooser:
- Added required prop to link chooser
- Added disabled prop to link chooser for input and checkbox
- Added validation state prop to link chooser
- Removed menuItems since they were causing focus ring bugs
- Added Guidelines Section
- Pressing the clear button with the chooser open will not close the chooser but will reset to default view
- Selecting a template/guideline resets list to locally stored so that next time it is opened it will display correct results
- Opening and closing the dropdown updates comboBoxState
- Types have been moved to types file
- Repository folder with duplicate functions has been reduced to 1 default function
- Converted to direct returns when possible
- Tests have been updated
- Menu is traversable with keyboard controls

### CODE REVIEW

File: src/components/LinkChooser/SearchResultSection.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+);
+
+const FetchingAnimation = () => (
+    <div
+        data-test-id="link-chooser-loader"
+        className="tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[350px]"
+    >
+        <img className="tw-w-[50px]" src={FetchingIcon} alt="Fetching" />
+    </div>
+);
```
Comment (SamCreasey) [inline src/components/LinkChooser/SearchResultSection.tsx:184]: just made into direct return

File: src/components/LinkChooser/SearchResultSection.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+const ImageMenuItem = ({ title, subtitle, preview }: ImageMenuItemProps) => (
```
Comment (SamCreasey) [inline src/components/LinkChooser/SearchResultSection.tsx:186]: just changed name

File: src/components/LinkChooser/NavigationMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            role="menuitem"
+            aria-label={`Navigate to ${section.id} section of search results.`}
+            key={section.id}
+            data-test-id="link-chooser-navigation-menu-item"
+            className={merge(["hover:tw-bg-black-0", isFocused && isFocusVisible && "tw-bg-black-0"])}
+        >
+            <MenuItem
+                title={title}
+                selectionIndicator={direction === "right" ? SelectionIndicatorIcon.CaretRight : undefined}
+                decorator={direction === "left" ? <IconArrowLeft /> : undefined}
```
Comment (SamuelAlev) [inline src/components/LinkChooser/NavigationMenu.tsx:69]:
```suggestion
                selectionIndicator={direction === "right" && SelectionIndicatorIcon.CaretRight}
                decorator={direction === "left" && <IconArrowLeft />}
```
Or typing don't allow that?

File: src/components/LinkChooser/SearchInput.tsx [hunk 4]
```diff
@@ -1,154 +1,112 @@
...
+                            disabled || !selectedResult ? "tw-text-black-40" : "tw-text-violet-60",
                         ])}
                         data-test-id="link-chooser-decorator-icon"
                     >
                         {decorator}
                     </div>
                 )}
                 <input
-                    id={useMemoizedId(propId)}
+                    id={useMemoizedId()}
```
Comment (SamuelAlev) [inline src/components/LinkChooser/SearchInput.tsx:63]: I don't think it is needed anymore as it should be contained in `ariaProps`

File: src/components/LinkChooser/mock/guidelines.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconLabel } from "../types";
+
+export const guidelines = [
+    {
+        id: "15",
+        preview:
+            "https://images.frontify.test/s3/frontify-dev-files/eyJwYXRoIjoibXNpcmljXC9hY2NvdW50c1wvYzRcLzFcL3Byb2plY3RzXC8yXC9hc3NldHNcLzhmXC83XC85MWU4MzdjZDk2YWQzN2EwYWY4OTZhYzA5OWRlNzIyNi0xNjM0MzAxNDUxLmpwZyJ9:msiric:Li3v_e6ogrxBoH-kli3whbqekW1K4ar5Dshif80q2o0?width=2400&height={height}",
```
Comment (SamuelAlev) [inline src/components/LinkChooser/mock/guidelines.ts:9]: `images.frontify.test` is not working outside of your local env

**Review by SamuelAlev — COMMENTED** [review]
Could you take a look to [the code smells](https://sonarcloud.io/project/issues?id=Frontify_arcade&pullRequest=530&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

Did some smoke test and looks good, I guess issues will raise up when we wire it to the endpoints

File: src/components/LinkChooser/NavigationMenu.tsx (hunk 3 above)
Comment (SamCreasey) [reply src/components/LinkChooser/NavigationMenu.tsx:69]: Typing did not allow this

File: src/components/LinkChooser/SearchInput.tsx (hunk 4 above)
Comment (SamCreasey) [reply src/components/LinkChooser/SearchInput.tsx]: removed

File: src/components/LinkChooser/mock/guidelines.ts (hunk 5 above)
Comment (SamCreasey) [reply src/components/LinkChooser/mock/guidelines.ts]: fixed

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**